pkg/pub_package_reader/lib/pub_package_reader.dart

```diff
@@ -718,3 +719,29 @@ Iterable<ArchiveIssue> checkScreenshots(
     yield* validateZalgo('screenshot description', s.description);
   }
 }
+
+Iterable<ArchiveIssue> checkFunding(String pubspecContent) sync* {
+  final map = loadYaml(pubspecContent) as Map;
+  if (!map.containsKey('funding')) {
+    return;
+  }
+  final funding = map['funding'];
+  if (funding == null || funding is! List || funding.isEmpty) {
+    yield ArchiveIssue(
+        '`pubspec.yaml` has invalid `funding`: only a list of URLs are allowed.');
+    return;
+  }
+  for (final item in funding) {
+    if (item is! String || item.trim().isEmpty) {
+      yield ArchiveIssue(
+          'Invalid `funding` value (`$item`): only URLs are allowed.');
+      continue;
+    }
+    final uri = Uri.tryParse(item.trim());
+    if (uri == null || uri.scheme != 'https') {
+      yield ArchiveIssue(
+          'Invalid `funding` value (`$item`): only `https` URLs are allowed.');
+      continue;
+    }
```

Comment:
For sanity let's also limit the length of the URL. Let's just say no more than 255 characters or something like that?

We don't want garbage in here...